Controllers/CustomersController.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class CustomersController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CustomersController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Customers
+        public async Task<IActionResult> Index()
+        {
+            var Customers = await _context.Customers
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .ToListAsync();
+
+            var customersVM = new List<CustomerFormViewModel>();
+
+            foreach (var customer in Customers)
+            {
+                customersVM.Add(_mapper.Map<CustomerFormViewModel>(customer));
+            }
+
+            return View(customersVM);
+        }
+
+        // GET: Customers by Entity Id
+        public async Task<IActionResult> ListCustomers(int? id)
+        {
```

Comment:
You need to write here logic that will hander case when `id=null`

Reply:
For an entity without customers, an empty is displayed as-is.

TO BE IMPLEMENTED: An warning message informing that the entity does not contain any customer.

Reply:
From the entity page, the user can browse the Customers list.
A warning message was applied to the views in case of an empty list.

Reply:
My mistake the null handling has nothing to do with an empty list. 
Null handling will be added to the controllers.

Reply:
Null handlings were added.